Extension/lib/content-script/preload.js

```diff
@@ -211,6 +211,10 @@
             }
         }
 
+        // Override window.WebSocket running the function from websocket.js
+        // https://github.com/AdguardTeam/AdguardBrowserExtension/issues/203
+        scriptsToApply.unshift('(' + overrideWebSocket.toString() + ')();');

```

Comment:
One more important note. We **must** inject this script inside `init` function, not like you do it now.

The purpose is to execute it **as soon as possible**, otherwise page may already create a WebSocket.